src/tiledb/cloud/utilities/_common.py

```diff
@@ -200,3 +202,44 @@ def max_memory_usage() -> int:
         result = 0
 
     return result
+
+
+def process_stream(
+    uri: str,
+    cmd: str,
+    read_size: int = 16 << 20,
+) -> Tuple[str, str]:
+    """
+    Process a stream of data from VFS with a subprocess.
+
+    If the file is large and the subprocess only reads a small amount of data,
+    then reduce `read_size` to improve performance.
+
+    :param uri: file URI
+    :param cmd: command to run in the subprocess
+    :param read_size: number of bytes to read per iteration, defaults to 16 MiB
+    :return: stdout and stderr from the subprocess
+    """
+
+    with tiledb.VFS().open(uri) as fp:
+        with subprocess.Popen(
+            cmd.split(),
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        ) as process:
+            while data := fp.read(read_size):
+                try:
+                    process.stdin.write(data)
+                except BrokenPipeError:
+                    # The subprocess has exited, stop reading
+                    break
```

Comment:
Is this a sign of a premature exit or error condition on whatever program you’re running?

Reply:
Also an expected early exit. For example, if we're streaming a 2G file to a program that only reads the first 10k of a file, it will exit early and we can stop streaming. `bcftools` does this when reading the sample name from a VCF file.